checks/binary_artifact.go

```diff
@@ -50,5 +49,5 @@ func BinaryArtifacts(c *checker.CheckRequest) checker.CheckResult {
 	}
 
 	// Return the score evaluation.
-	return evaluation.BinaryArtifacts(CheckBinaryArtifacts, c.Dlogger, &rawData)
+	return evaluation.BinaryArtifacts(CheckBinaryArtifacts, c.Dlogger, &rawData, c.RepoClient)
```

Comment:
I know the change is well intentioned - really appreciate the effort. But I don't think we want make that change unfortunately. I should have told you earlier.

I know you're trying to move the evaluation of the results into the evaluation (which is what we want long-term), but the evaluation should *not* require any further access to the API: The raw results should contain all the information necessary to evaluate the results; it's the main reason we created them. For example, imagine someone retrieves them from our BQTable, they should be able to parse and evaluation using their "policy".

There's no simple solution today. You can think about where this would live in the raw structure https://github.com/ossf/scorecard/blob/main/checker/raw_result.go. We have the same limitation today for permissions: we have "exceptions" in the sense that we allow write permission if a certain action is used (e.g., create a release needs `content:write` permission). We would like to move them to the evaluation, but we have not decided which structure we should keep that data in.

If you have some ideas where this information could live in the raw results: new structure or in an existing structure, that'd be really nice. But for this PR I think it's fine to create the exception in the `raw/` instead.

Does this make sense?

/cc @raghavkaul something to be aware that we need to fix.

Reply:
Maybe it could live under a generic `Workflows` structure with a list actions found for each job. Ideally that would also contain permission information, so that both the permission job, dangerous workflow and permission checks could use it. But it's a pretty big change so we need to think about it a bit more.

Reply:
That makes sense, thanks for the explanation!